File: server/microservices/Socigy.Microservices.Content/Requests/Content/RecommendationRequest.cs
using Socigy.Microservices.Content.Enums;

namespace Socigy.Microservices.Content.Requests.Content
{
    public class RecommendationRequest
    {
        public List<ContentType>? ContentTypes { get; set; }
        public Guid? ContentProfile { get; set; }
        public DateTime? PostedBefore { get; set; }
        public DateTime? PostedAfter { get; set; }
        public List<Guid>? CreatorIds { get; set; }
        public List<Guid>? CategoryIds { get; set; }
        public List<Guid>? InterestIds { get; set; }
        public List<Guid>? ExcludedCreatorIds { get; set; }
        public List<Guid>? ExcludedCategoryIds { get; set; }
        public List<Guid>? ExcludedInterestIds { get; set; }
        public int Limit { get; set; } = 10;
        public int Offset { get; set; } = 0;

        public string? Search { get; set; }
        public Guid? TargetUserId { get; set; }
    }
}

File: client/native/app/lib/structures/content/posts/RecommendationRequest.ts
import { Guid } from "../../Guid";

export enum ContentType {
  Take,
  Frame,
  FullFrame,
  Music,
  Podcast,
  Quote,
  Discussion,
  Blog,
  News,
  Poll,
  Video,
  LiveTake,
  LivePodcast,
  Stream,
}

export interface RecommendationRequest {
  contentTypes?: ContentType[];

  contentProfile?: Guid;

  postedBefore?: Date;
  postedAfter?: Date;

  creatorIds?: Guid[];
  categoryIds?: Guid[];
  interestIds?: Guid[];
  excludedCreatorIds?: Guid[];
  excludedCategoryIds?: Guid[];
  excludedInterestIds?: Guid[];

  limit: number;
  offset: number;

  search?: string;
  targetUserId?: Guid;
}
